bridges/YoutubeBridge.php

```diff
@@ -176,10 +183,16 @@ public function collectData(){
 			}
 		} elseif($this->getInput('p')) { /* playlist mode */
 			$this->request = $this->getInput('p');
+			$url_feed = self::URI . 'feeds/videos.xml?playlist_id=' . urlencode($this->request);
 			$url_listing = self::URI . 'playlist?list=' . urlencode($this->request);
 			$html = $this->ytGetSimpleHTMLDOM($url_listing)
 				or returnServerError("Could not request YouTube. Tried:\n - $url_listing");
-			$this->ytBridgeParseHtmlListing($html, 'tr.pl-video', '.pl-video-title a');
+			$item_count = $this->ytCountItemsFromHtmlListing($html, 'tr.pl-video', '.pl-video-title a');
+			if ($item_count <= 15 && ($xml = $this->ytGetSimpleHTMLDOM($url_feed))) {
+				$this->ytBridgeParseXmlFeed($xml);
+			} else {
+				$this->ytBridgeParseHtmlListing($html, 'tr.pl-video', '.pl-video-title a');
+			}
```

Comment:
Based on how the user/channel mode is implemented, there are cases where the xml feed is not available. In those situations I suggest a fallback to HTML and if that doesn't work an error message. Please have a look at how this is implemented for the user/channel mode:

https://github.com/RSS-Bridge/rss-bridge/blob/8ba817478bd1700ec40172b3fdbca9b5dd37314d/bridges/YoutubeBridge.php#L170-L176